pandas/_libs/tslibs/period.pyx

```diff
@@ -2404,6 +2406,13 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, reso = parse_time_string(value, freq)
+            try:
+                ts = Timestamp(value, freq=freq)
+                nanosecond = ts.nanosecond
```

Comment:
this could be in an `else` clause

Reply:
Sorry, I don't get this point.
I go through the `Timestamp` constructor just to obtain the nanosecond value. Note that it could be a performance issue !

Reply:
else in a try/except is how you do things *after* the try succeeds

Reply:
Got it. The last PR should be fine.

First point to notice, I want to make it super clear about performance of this PR.

Using version 1.1.0.dev0+1823.g8a559cb07:
```
%timeit p = pd.Period('2020-12-31 12:00:00.123456789')
784 µs ± 41.7 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Version 1.0.4:
```
%timeit p = pd.Period('2020-12-31 12:00:00.123456789')
673 µs ± 28.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Afterwards, it does not look dramatic, but I wanted to notice it.

Second point, As far I've seen, the `dateutil` and `datetime` are not planning to support nanosecond yet. This solution could be here for a while.

Reply:
could you avoid some overhead by conditioning on `reso`?  i.e. it should only be relevant when `reso == "microsecond"`?